client/omnichannel/agentManager/AgentsManagerTable.js

```diff
@@ -0,0 +1,113 @@
+		<TextInput flexShrink={0} placeholder={t('Search')} addon={<Icon name='magnifier' size='x20'/>} onChange={handleChange} value={text} />
+	</Box>;
+};
+
+const sortDir = (sortDir) => (sortDir === 'asc' ? 1 : -1);
+
+const useQuery = ({ text, itemsPerPage, current }, [column, direction], cache) => useMemo(() => ({
+	fields: JSON.stringify({ name: 1, username: 1, emails: 1, avatarETag: 1 }),
+	query: JSON.stringify({
+		$or: [
+			{ 'emails.address': { $regex: text || '', $options: 'i' } },
+			{ username: { $regex: text || '', $options: 'i' } },
+			{ name: { $regex: text || '', $options: 'i' } },
+		],
+	}),
+	sort: JSON.stringify({ [column]: sortDir(direction), usernames: column === 'name' ? sortDir(direction) : undefined }),
+	...itemsPerPage && { count: itemsPerPage },
+	...current && { offset: current },
+}), [text, itemsPerPage, current, column, direction, cache]);
+
+export function RemoveButton({ _id, onChange, type }) {
+	const deleteAction = useEndpointAction('DELETE', `livechat/users/${ type }/${ _id }`);
+
+	const handleRemoveClick = useCallback(async () => {
+		const result = await deleteAction();
+		if (result.success === true) {
+			onChange();
+		}
+	}, [deleteAction, onChange]);
+
+	return <Table.Cell fontScale='p1' color='hint' onClick={handleRemoveClick} style={style}><Icon name='trash' size='x20'/></Table.Cell>;
+}
+
+export function AgentsManagerTable({ cache, setCache, type }) {
+	const t = useTranslation();
+
+	const [params, setParams] = useState({ text: '', current: 0, itemsPerPage: 25 });
+	const [sort, setSort] = useState(['name', 'asc']);
+
+	const onChange = useCallback(() => {
+		setCache(new Date());
+	}, [setCache]);
+
+
+	const debouncedParams = useDebouncedValue(params, 500);
+	const debouncedSort = useDebouncedValue(sort, 500);
+	const query = useQuery(debouncedParams, debouncedSort, cache);
+
+	const data = useEndpointData(`livechat/users/${ type }`, query) || {};
+
+	const onHeaderClick = useCallback((id) => {
+		const [sortBy, sortDirection] = sort;
+
+		if (sortBy === id) {
+			setSort([id, sortDirection === 'asc' ? 'desc' : 'asc']);
+			return;
+		}
+		setSort([id, 'asc']);
+	}, [sort]);
+
+	const mediaQuery = useMediaQuery('(min-width: 1024px)');
+
+	const header = useMemo(() => [
+		<Th key={'name'} direction={sort[1]} active={sort[0] === 'name'} onClick={onHeaderClick} sort='name' w='x200'>{t('Name')}</Th>,
+		mediaQuery && <Th key={'username'} direction={sort[1]} active={sort[0] === 'username'} onClick={onHeaderClick} sort='username' w='x140'>{t('Username')}</Th>,
+		<Th key={'email'} direction={sort[1]} active={sort[0] === 'emails.adress'} onClick={onHeaderClick} sort='emails.address' w='x120'>{t('Email')}</Th>,
+		<Th key={'remove'} w='x40'>{t('Remove')}</Th>,
+	].filter(Boolean), [sort, onHeaderClick, t, mediaQuery]);
+
+	const renderRow = useCallback(({ emails, _id, username, name, avatarETag }) => <Table.Row key={_id} tabIndex={0} role='link' action qa-user-id={_id}>
```

Comment:
In this case, it's better if you create a `RenderAgentRow` component, and use `React.memo`. Then you pass it as a GenericTable children, instead of `renderRow` prop.